### PR TITLE
Document all generator types in one place.

### PR DESCRIPTION
**Pull request by sigilante**

### CODE REVIEW

**Comment by sigilante** [comment]
Resolves #343.

File: content/guides/additional/generators.md [hunk 1]
```diff
@@ -0,0 +1,240 @@
...
+            ==
+            ~
+        ==
+    ==
+```
+
+The Dojo will modify the sample by inserting `%~` (constant null) at the end of each collection, since the Dojo adapts
+the input arguments into a list (either the unnamed/required argument list or the named/optional argument list).
+
+### Zero argumetns
```
Comment (tinnus-napbus) [inline content/guides/additional/generators.md:108]:
```suggestion
### Zero arguments
```

File: content/guides/additional/generators.md [hunk 2]
```diff
@@ -0,0 +1,240 @@
...
+
+`/gen/vats.hoon` is commonly used to check on the status of installed desks.  It can be invoked with optional arguments:
+
+```hoon
```
Comment (tinnus-napbus) [inline content/guides/additional/generators.md:112]:
````suggestion
```
````

File: content/guides/additional/generators.md [hunk 3]
```diff
@@ -0,0 +1,240 @@
...
+and now expects additional input.  Let's give it an answer:
+
+```hoon
+: color: red
+Into the Gorge of Eternal Peril with you!
+Aaaaagh!
+```
+
+`%ask` generators return `sole-effect`s.  For more information on these, consult the [guide on command-line agents](/guides
+/additional/cli).
```
Comment (tinnus-napbus) [inline content/guides/additional/generators.md:227]:
```suggestion
`%ask` generators return `sole-effect`s.  For more information on these, consult the [guide on command-line agents](/guides/additional/cli).
```

**Review by tinnus-napbus — APPROVED** [review]

**Pull request merged**